ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh

```diff
@@ -55,20 +55,55 @@ echo "Get install-gather, if there is one..."
 cp /root/dev-scripts/ocp/*/log-bundle*.tar.gz /tmp/artifacts/log-bundle-\$HOSTNAME.tar.gz || true
 
 echo "Get sosreport including sar data..."
-sos report --case-id "\$HOSTNAME" --batch -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh,yum --tmp-dir /tmp/artifacts
+sos report --case-id "\$HOSTNAME" --batch \
+  -o container_log,filesys,kvm,libvirt,logs,networkmanager,podman,processor,rpm,sar,virsh,yum \
+  -k podman.all -k podman.logs \
+  --tmp-dir /tmp/artifacts
 
 echo "Get libvirt logs..."
 tar -czC "/var/log/libvirt/qemu" -f "/tmp/artifacts/libvirt-logs-\$HOSTNAME.tar.gz" --transform "s?^\.?libvirt-logs-\$HOSTNAME?" .
 
+. common.sh
+. network.sh
+. utils.sh
+
+# Pass master IPs to installer-gather script to collect info from nodes which didn't join the cluster
+NODE_IPS=()
+for (( n=0; n<\$NUM_MASTERS; n++ ))
+do
+    node_name=\$(printf \$MASTER_HOSTNAME_FORMAT \$n)
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//ip/dhcp[*]/host[@name='\${node_name}']/@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+for (( n=0; n<\$NUM_WORKERS; n++ ))
+do
+    node_name=\$(printf \$WORKER_HOSTNAME_FORMAT \$n)
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+for (( n=0; n<\$NUM_EXTRA_WORKERS; n++ ))
+do
+    node_name="extraworker-%d"
+    node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+    NODE_IPS+=("\$node_ip")
+done
+
+# Collect sos report from each known node
+for NODE_IP in "\${NODE_IPS[@]}"; do
+  ssh "${SSHOPTS[@]}" core@\$(wrap_if_ipv6 \$NODE_IP) \
+    sos report --case-id "\$HOSTNAME" \
+      -o container_log,filesys,logs,networkmanager,podman,processor,sar \
+      -k podman.all -k podman.logs \
+      --tmp-dir /tmp/artifacts && \
+  scp "${SSHOPTS[@]}" core@\$(wrap_if_ipv6 \$NODE_IP):/tmp/artifacts /tmp/artifacts/ || true
```

Comment:
I think ``SSHOPTS[@]`` might be using the packet's connection info and not the libvirt host connection info
Maybe we need to define ``INTERNAL_SSHOPTS="-i /root/.ssh/id_rsa ...`` (don't quote me on the SSH key location) and use it